test/test_linalg.py

```diff
@@ -31,6 +33,125 @@ def fn(a, b):
         self.assertEqual(actual, original)
         self.assertEqual(actual, scripted)
 
+    def test_norm_dtype(self, device):
+        def run_test_case(input_size, ord, keepdim, from_dtype, to_dtype, compare_dtype):
+            msg = (
+                f'input_size={input_size}, ord={ord}, keepdim={keepdim}, '
+                f'from_dtype={from_dtype}, to_dtype={to_dtype}')
+            input = torch.randn(*input_size, dtype=from_dtype, device=device)
+            result = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=from_dtype)
+            self.assertEqual(result.dtype, from_dtype, msg=msg)
+            result_converted = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=to_dtype)
+            self.assertEqual(result_converted.dtype, to_dtype, msg=msg)
+            self.assertEqual(result.to(compare_dtype), result_converted.to(compare_dtype), msg=msg)
+
+            result_out_converted = torch.empty_like(result_converted)
+            torch.linalg.norm(input, ord, keepdim=keepdim, dtype=to_dtype, out=result_out_converted)
+            self.assertEqual(result_out_converted.dtype, to_dtype, msg=msg)
+            self.assertEqual(result_converted, result_out_converted, msg=msg)
+
+        ord_vector = [0, 1, -1, 2, -2, 3, -3, 4.5, -4.5, inf, -inf, None]
+        ord_matrix = [1, -1, 2, -2, inf, -inf, None]
+        S = 10
+        test_cases = [
+            ((S, ), ord_vector),
+            ((S, S), ord_matrix),
+        ]
+        for keepdim in [True, False]:
+            for input_size, ord_settings in test_cases:
+                for ord in ord_settings:
+                    # float to double
+                    run_test_case(input_size, ord, keepdim, torch.float, torch.double, torch.float)
+                    # double to float
+                    run_test_case(input_size, ord, keepdim, torch.double, torch.double, torch.float)
+
+        # TODO: Once dtype arg is supported in nuclear and frobenius norms, remove the following test
```

Comment:
I feel like it would be best to fix this in a follow up PR, but let me know if it would be preferable to include it in this one.

Reply:
This PR's scope is already a little too big (see note about cutting named tensor support).